ncm-openstack/src/main/pan/components/openstack/neutron.pan

```diff
@@ -0,0 +1,147 @@
+# ${license-info}
+# ${developer-info}
+# ${author-info}
+
+
+declaration template components/openstack/neutron;
+
+@documentation {
+    The Neutron configuration options in ml2_conf.ini "ml2" Section.
+}
+type openstack_neutron_ml2 = {
+    @{WARNING: After you configure the ML2 plug-in,
+    removing values in the type_drivers option can lead to database inconsistency}
+    'type_drivers' : type_neutrondriver[] = list('flat', 'vlan', 'vxlan')
+    @{Ordered list of network_types to allocate as tenant networks. The default
+    value "local" is useful for single-box testing but provides no connectivity
+    between hosts}
+    'tenant_network_types' : type_neutrondriver[] = list('vxlan')
+    @{An ordered list of networking mechanism driver entrypoints to be loaded from
+    the neutron.ml2.mechanism_drivers namespace}
+    'mechanism_drivers' : string[] = list('linuxbridge', 'l2population')
+    @{An ordered list of extension driver entrypoints to be loaded from the
+    neutron.ml2.extension_drivers namespace}
+    'extension_drivers' : type_neutronextension[] = list('port_security')
+} = dict();
+
+@documentation {
+    The Neutron configuration options in ml2_conf.ini "ml2_type_flat" Section.
+}
+type openstack_neutron_ml2_type_flat = {
+    @{List of physical_network names with which flat networks can be created. Use
+    default "*" to allow flat networks with arbitrary physical_network names. Use
+    an empty list to disable flat networks}
+    'flat_networks' : string[] = list('provider')
+} = dict();
+
+@documentation {
+    The Neutron configuration options in ml2_conf.ini "ml2_type_vxlan" Section.
+}
+type openstack_neutron_ml2_type_vxlan = {
+    @{Configure the VXLAN network identifier range for self-service networks}
+    'vni_ranges' : string = '1:1000'
```

Comment:
this needs its own type `type_port_range = string with match(SELF, "\d+:\d+")`, also in template-library-core types.pan